src/control/dmg/service.go

```diff
@@ -42,20 +42,24 @@ type KillRankSvcCmd struct {
 	PoolUUID string `short:"p" long:"pool-uuid" description:"Pool uuid that rank relates to"`
 }
 
-// Execute is run when KillRankSvcCmd activates
-func (k *KillRankSvcCmd) Execute(args []string) error {
-	if err := connectHosts(); err != nil {
-		return errors.WithMessage(err, "unable to connect to hosts")
-	}
-
-	// run kill rank command with specified parameters on all connected servers
-	errors := conns.KillRank(k.PoolUUID, k.Rank)
+// run kill rank command with specified parameters on all connected servers
+func killRankSvc(uuid string, rank uint32) {
+	errors := conns.KillRank(uuid, rank)
 
 	if len(errors) == 0 {
 		fmt.Println("Kill Rank succeeding on all active connections!")
 	} else {
 		fmt.Printf(unpackFormat(errors), "Kill Rank command failures")
 	}
+}
+
+// Execute is run when KillRankSvcCmd activates
+func (k *KillRankSvcCmd) Execute(args []string) error {
+	if err := connectHosts(); err != nil {
+		return errors.WithMessage(err, "unable to connect to hosts")
+	}
+
+	killRankSvc(k.PoolUUID, k.Rank)
 
 	// exit immediately to avoid continuation of main
 	os.Exit(0)
```

Comment:
Was just reminded by looking at #280 - Do we want to exit with a nonzero error if something goes wrong?

Reply:
yes I was thinking about this and decided that in this case the exit code should be to do with application failure (therefore if all commands get sent and received it's a success regardless of server side results). I'm happy to change that and maybe we have different exit codes for each. possibly a discussion for another patch though?

Reply:
Yeah, I'm okay with deferring to a future discussion.